router/router.go

```diff
@@ -66,21 +66,25 @@ type HandleDestOAuthRespParamsT struct {
 
 // HandleT is the handle to this module.
 type HandleT struct {
-	responseQ                              chan jobResponseT
-	jobsDB                                 jobsdb.MultiTenantJobsDB
-	errorDB                                jobsdb.JobsDB
-	netHandle                              NetHandleI
-	MultitenantI                           tenantStats
-	destName                               string
-	workers                                []*workerT
-	perfStats                              *misc.PerfStats
-	successCount                           uint64
-	failCount                              uint64
-	failedEventsListMutex                  sync.RWMutex
-	failedEventsList                       *list.List
-	failedEventsChan                       chan jobsdb.JobStatusT
-	toClearFailJobIDMutex                  sync.Mutex
-	toClearFailJobIDMap                    map[int][]string
+	responseQ             chan jobResponseT
+	jobsDB                jobsdb.MultiTenantJobsDB
+	errorDB               jobsdb.JobsDB
+	netHandle             NetHandleI
+	MultitenantI          tenantStats
+	destName              string
+	workers               []*workerT
+	perfStats             *misc.PerfStats
+	successCount          uint64
+	failCount             uint64
+	failedEventsListMutex sync.RWMutex
+	failedEventsList      *list.List
+	failedEventsChan      chan jobsdb.JobStatusT
+
+	eventOrderSyncMu        sync.Mutex // protects all maps below
+	toClearFailJobIDMap     map[int][]string
+	toClearAbortedJobIDMap  map[int][]string
+	toRemoveAbortedJobIDMap map[int]map[string]map[int64]struct{}
```

Comment:
Can you add a small description regarding the purpose of those maps?

Reply:
no description will be good enough, but yes :)

Reply:
The difference between Clear and Remove is semantically very close. To understand the difference, you need to go back and forth inside the code. This is why I am suggesting a short explanation here. Ofc with the new abstraction, that won't be a problem.